test/e2e_node/eviction_test.go

```diff
@@ -520,19 +520,34 @@ func runEvictionTest(f *framework.Framework, pressureTimeout time.Duration, expe
 		})
 
 		ginkgo.AfterEach(func() {
-			defer func() {
+			prePullImagesIfNeccecary := func() {
 				if expectedNodeCondition == v1.NodeDiskPressure && framework.TestContext.PrepullImages {
 					// The disk eviction test may cause the prepulled images to be evicted,
 					// prepull those images again to ensure this test not affect following tests.
 					PrePullAllImages()
 				}
-			}()
+			}
+			defer prePullImagesIfNeccecary()
+
 			ginkgo.By("deleting pods")
 			for _, spec := range testSpecs {
 				ginkgo.By(fmt.Sprintf("deleting pod: %s", spec.pod.Name))
 				f.PodClient().DeleteSync(spec.pod.Name, &metav1.DeleteOptions{}, 10*time.Minute)
 			}
+
+			// In case the test fail, we should make sure not contidion no longer exist after the test runs
```

Comment:
Nit: Mind rewriting this comment? It has some grammar/spelling errors which make it a little difficult to understand.

Reply:
Thanks for catching it! Updated the PR now.